### DISCUSSION TITLE
실행 컨텍스트가 가비지 컬렉션과 메모리에 어떤 영향을 미치는지, 이를 최적화하기 위한 전략은 무엇인가요?

### DISCUSSION THREAD

**Discussion by lledellebell** [diving-javascript/you-dont-know-js-yet#29] [실행 컨텍스트(Execution Context)]
- #18

**Comment by wogus216** [comment]
실행컨텍스트는 메모리 할당을 관리하는 구조이며, 이는 가비지 컬렉션의 대상
가비지 컬렉션은 프로그램이 더 이상 사용하지 않는 메모리를 자동으로 회수하는 과정

실행 컨텍스트 생성과 소멸
각 함수 호출 시 새로운 실행 컨텍스트가 스택에 푸시되며, 함수 실행이 완료되면 해당 컨텍스트는 스택에서 팝됩니다.
실행 컨텍스트가 스택에서 제거되면 그 안에 포함된 모든 지역 변수와 함수 선언 등도 메모리에서 해제될 수 있습니다. 단, 클로저가 있을 경우 해당 변수는 여전히 접근 가능하며 메모리에 남아 있게 됩니다.

메모리 관리 최적화 전략

변수의 적절한 생명주기 관리: 필요한 최소 범위에서만 선언하고 사용
클로저 사용 최소화: 사용하지 않는 데이터를 메모리에 게속 유지하기에 신중하게 사용 클로저가 필요없어지면 null이나 nudefined 로 설정하여 참조를 명시적으로 끊어주는 것이 좋음
이벤트 리스너 제거: 이벤트 리스너는 DOM요소와 같이 큰 객체를 참조 할 수 있으며, 이는 메모리의 누수의 원인이 될 수 있음
대규모 데이터 구조 관리: 큰 배열이나 겍체등은 사용 후 적절히 정리하거나 필요한 부분만 추출하여 저장
실행 컨텍스트와 메모리 관리를 최적화하면 애플리케이션의 성능과 반응성을 크게 향상 가능
